includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -1489,4 +1577,86 @@ private function get_address_data_for_payment_request( $order ) {
 		];
 	}
 
+	/**
+	 * Collects the payment information needed for processing a payment intent.
+	 *
+	 * @param WC_Order $order The WC Order to be paid for.
+	 * @return array An array containing the payment information for processing a payment intent.
+	 */
+	private function prepare_payment_information_from_request( WC_Order $order ) {
+		// TODO: throw exception if any required information is missing.
+
+		$payment_method               = sanitize_text_field( wp_unslash( $_POST['wc-stripe-payment-method'] ?? '' ) ); // phpcs:ignore WordPress.Security.NonceVerification.Missing
+		$selected_payment_type        = sanitize_text_field( wp_unslash( $_POST['wc_stripe_selected_upe_payment_type'] ?? '' ) ); // phpcs:ignore WordPress.Security.NonceVerification.Missing
+		$capture_method               = empty( $this->get_option( 'capture' ) ) || $this->get_option( 'capture' ) === 'yes' ? 'automatic' : 'manual'; // automatic | manual.
+		$save_payment_method_to_store = isset( $_POST['save_payment_method'] ) ? 'yes' === wc_clean( wp_unslash( $_POST['save_payment_method'] ) ) : false;
+		$currency                     = strtolower( get_woocommerce_currency() );
```

Comment:
```suggestion
		$currency                     = strtolower( $order->get_currency() );
```

I think this should ise the orders currency not the stores currency, is that correct?

Reply:
Nice catch! Yes, I think you're right.

I followed the [old implementation](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/class-wc-stripe-intent-controller.php#L338) when creating an intent where the store currency is used, but we didn't have an order at that point with legacy PE. But we do have an order in this case.

Thanksss